core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -520,37 +488,18 @@ public String toString() {
       }
    }
 
-   private static class ExceptionCollector<T> extends CompletableFuture<T> implements Collector<T> {
-      ExceptionCollector(Throwable throwable) {
-         completeExceptionally(throwable);
-      }
-
-      @Override
-      public CompletableFuture<T> getFuture() {
-         return this;
-      }
-
-      @Override
-      public void primaryException(Throwable throwable) {
-         throw new UnsupportedOperationException();
-      }
-
-      @Override
-      public void primaryResult(T result, boolean success) {
-         throw new UnsupportedOperationException();
-      }
-   }
+   private static class CombiningCollector implements BiasedCollector, Function<Void, CompletableFuture<ValidResponse>> {
```

Comment:
I'd like a bit of javadoc here explaining what is being combined, at a first glance it looks like it just wraps a `MultiTargetCollectorImpl`